<launch>
    <!--
        Indicates if ROS is running on a physical SCUTTLE (value = true) or a virtual one
        (value = false), e.g. when running in Gazebo.
    -->
    <arg name="is_physical" default="true"/>

    <!--
        When we're in Gazebo ...
    -->
    <group unless="$(arg is_physical)">
        <!--
            Start the translation node that translates Gazebo contact sensor responses to
            scuttle bumper responses
        -->
        <!-- Left hand front bumper -->
        <node
            name="gazebo_translator_front_left"
            ns="scuttle/sensor/bumper"
            pkg="scuttle_bumper"
            respawn="true"
            type="gazebo_contact_sensor_translator.py"
        >
            <param name="bumper_frame_id" value="front_bumper_plate_left_link" />
            <param name="bumper_name" value="front-left" />
            <param name="bumper_section_width" type="double" value="0.230" />
            <param name="bumper_section_height" type="double" value="0.080" />
            <param name="gazebo_contact_debounce_time_in_seconds" type="double" value="0.3" />
            <!-- This value should match the value in the `bumper.gazebo` URDF file -->
            <param name="gazebo_contact_sample_frequency_in_hz" type="int" value="15" />

            <!-- Remap the subscription topic to point to the topic for the front left bumper -->
            <remap from="/scuttle/sensor/bumper/gazebo" to="front_left_gazebo" />
            <!-- Remap the publishing topic -->
            <remap from="/scuttle/sensor/bumper/events" to="events" />
        </node>

        <!-- Right hand front bumper -->
        <node
            name="gazebo_translator_front_right"
            ns="scuttle/sensor/bumper"
            pkg="scuttle_bumper"
            respawn="true"
            type="gazebo_contact_sensor_translator.py"
        >
            <param name="bumper_frame_id" value="front_bumper_plate_right_link" />
            <param name="bumper_name" value="front-right" />
            <param name="bumper_section_width" type="double" value="0.230" />
            <param name="bumper_section_height" type="double" value="0.080" />
            <param name="gazebo_contact_debounce_time_in_seconds" type="double" value="0.3" />
            <!-- This value should match the value in the `bumper.gazebo` URDF file -->
            <param name="gazebo_contact_sample_frequency_in_hz" type="int" value="15" />

            <remap from="/scuttle/sensor/bumper/gazebo" to="front_right_gazebo" />
            <!-- Remap the publishing topic -->
            <remap from="/scuttle/sensor/bumper/events" to="events" />
        </node>
    </group>

    <group if="$(arg is_physical)">
        <!--
            Start the bumper node that connects to the RPi GPIO pins to listen for

        -->
    </group>
</launch>